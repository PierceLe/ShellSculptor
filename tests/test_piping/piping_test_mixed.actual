>> cd ~/test_data
cd: no such file or directory: /Users/piercele/test_data
>> cat assignment_data1001.txt | wc -l
cat: assignment_data1001.txt: No such file or directory
       0
>> cat assignment_data1001.txt | grep "pets" | head -n 10
cat: assignment_data1001.txt: No such file or directory
>> cat assignment_data1001.txt | grep "fish" | sort
cat: assignment_data1001.txt: No such file or directory
>> cat assignment_data1001.txt | grep "dogs" | sort -r
cat: assignment_data1001.txt: No such file or directory
>> cat assignment_data1001.txt | tr ' ' '\n' | sort | uniq
cat: assignment_data1001.txt: No such file or directory
>> cat assignment_data1001.txt | tr ' ' '\n' | sort | uniq -c | sort -nr | head -n 10
cat: assignment_data1001.txt: No such file or directory

