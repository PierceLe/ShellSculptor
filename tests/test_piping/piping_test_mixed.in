cd ~/test_data
cat assignment_data1001.txt | wc -l
cat assignment_data1001.txt | grep "pets" | head -n 10
cat assignment_data1001.txt | grep "fish" | sort
cat assignment_data1001.txt | grep "dogs" | sort -r
cat assignment_data1001.txt | tr ' ' '\n' | sort | uniq
cat assignment_data1001.txt | tr ' ' '\n' | sort | uniq -c | sort -nr | head -n 10
