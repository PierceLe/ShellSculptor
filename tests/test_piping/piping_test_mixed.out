>> cd ~/test_data
>> cat assignment_data1001.txt | wc -l
27
>> cat assignment_data1001.txt | grep "pets" | head -n 10
The survey received 112 responses, with diffferent variables being represented in the rows, with columns showing the answer to these variables. We had 11 total questions, with 6 qualitative questions (Gender, type of pets, if money spent was worth it, how often you take pictures of pet, reason for having a pet, disadvantages of pets), and 5 quantitative questions (money spent, satisfaction of pets from 1-10, hours of free time, time spent with pets, and age of the owner). We then organised the data through using the “factor” code to group all the same answers together.
Certain absurd answers such as ADHD, when asked about the disadvantages of pets, were excluded manually. Response involving more obscured pets such as amphibians, arachnids, or horses were excluded since we want to investigate the relationship between more mainstream and popular type of pets, such as dogs and cats, rather than the more exotic types.
From the boxplots, we see that dogs and cats owners will often feel the most satisfaction from their pets. This is demonstrated in the average value of satisfaction of dogs and cats, with cats offering an average satisfaction of ~8.41 and dogs offering an average satisfaction of ~8.37 respectively. Guinea pigs will still offer a certain level of satisfaction with an average of ~6.25. Unfortunately, fish seems to offer the least level of satisfaction with an average of ~3.36.
In summary, the mean satisfaction score of 8.34 indicates that most participants feel highly satisfied with their pets, with 34 individuals giving a perfect rating of 10, signifying immense happiness in their pet companionship, while only 15 respondents rated below 5, demonstrating that a majority of participants genuinely enjoy their time with their pets.
In summary, there is a clear positive correlation in the data, showing that individuals with more free time per week tend to allocate approximately 28.5% of that time to spend with their pets, indicating that participants in the study highly value the time they share with their animal companions.
In summary, the data indicates that a majority of participants find pets to be a source of high satisfaction, with dog and cat owners typically reporting higher satisfaction scores, while fish and guinea pig owners tend to report lower scores. However, individual opinions vary, and there are exceptions within each category. Additionally, there is a positive correlation between the time spent with pets and the amount of free time individuals have, suggesting that people are willing to allocate more of their available free time to bond with their pets, emphasizing the value of pet companionship in their lives.
What are the advantages and disadvantages of each type of pets?
The main reason for why the owners keep their pet is because of pets cute appearance! While the reason from most fish owners is that fishes benefit their mental health. What comes to the next is many pet owners believe that the company from pet will solve their loneliness. Besides, dogs have all-round advantages compared with other anim
According to our graph, the median of ratio between time spent and free time of dogs has the highest value, following by cat and guinea. Moreover, we can see that the threshold line is over the rate of 1.0, it means someone define their time for pets as different from free time.
Research Question 2 investigate the advantages and disadvantages of various types of pets based on various responses. The responses shows that fish are the lowest maintenance pet as the owner only have cleaning issues as their concerns. Dogs and cats are cited as a high maintenance pet as it comes with many challenges such as financial issues. The main reason to keep a pet will be due to its cute appearance, while fish are seen to be improving mental health. The report also showed that dogs owner tend to spend more of their free time with their dogs, suggesting a strong commitment to dog care.
>> cat assignment_data1001.txt | grep "fish" | sort
According to the graph above, fish would be the easiest pet to take care, which has only one type of disadvantages, while the others have multiple issues.
From the boxplots, we see that dogs and cats owners will often feel the most satisfaction from their pets. This is demonstrated in the average value of satisfaction of dogs and cats, with cats offering an average satisfaction of ~8.41 and dogs offering an average satisfaction of ~8.37 respectively. Guinea pigs will still offer a certain level of satisfaction with an average of ~6.25. Unfortunately, fish seems to offer the least level of satisfaction with an average of ~3.36.
In summary, the data indicates that a majority of participants find pets to be a source of high satisfaction, with dog and cat owners typically reporting higher satisfaction scores, while fish and guinea pig owners tend to report lower scores. However, individual opinions vary, and there are exceptions within each category. Additionally, there is a positive correlation between the time spent with pets and the amount of free time individuals have, suggesting that people are willing to allocate more of their available free time to bond with their pets, emphasizing the value of pet companionship in their lives.
Research Question 2 investigate the advantages and disadvantages of various types of pets based on various responses. The responses shows that fish are the lowest maintenance pet as the owner only have cleaning issues as their concerns. Dogs and cats are cited as a high maintenance pet as it comes with many challenges such as financial issues. The main reason to keep a pet will be due to its cute appearance, while fish are seen to be improving mental health. The report also showed that dogs owner tend to spend more of their free time with their dogs, suggesting a strong commitment to dog care.
The main reason for why the owners keep their pet is because of pets cute appearance! While the reason from most fish owners is that fishes benefit their mental health. What comes to the next is many pet owners believe that the company from pet will solve their loneliness. Besides, dogs have all-round advantages compared with other anim
>> cat assignment_data1001.txt | grep "dogs" | sort -r
The main reason for why the owners keep their pet is because of pets cute appearance! While the reason from most fish owners is that fishes benefit their mental health. What comes to the next is many pet owners believe that the company from pet will solve their loneliness. Besides, dogs have all-round advantages compared with other anim
Research Question 2 investigate the advantages and disadvantages of various types of pets based on various responses. The responses shows that fish are the lowest maintenance pet as the owner only have cleaning issues as their concerns. Dogs and cats are cited as a high maintenance pet as it comes with many challenges such as financial issues. The main reason to keep a pet will be due to its cute appearance, while fish are seen to be improving mental health. The report also showed that dogs owner tend to spend more of their free time with their dogs, suggesting a strong commitment to dog care.
From the boxplots, we see that dogs and cats owners will often feel the most satisfaction from their pets. This is demonstrated in the average value of satisfaction of dogs and cats, with cats offering an average satisfaction of ~8.41 and dogs offering an average satisfaction of ~8.37 respectively. Guinea pigs will still offer a certain level of satisfaction with an average of ~6.25. Unfortunately, fish seems to offer the least level of satisfaction with an average of ~3.36.
Certain absurd answers such as ADHD, when asked about the disadvantages of pets, were excluded manually. Response involving more obscured pets such as amphibians, arachnids, or horses were excluded since we want to investigate the relationship between more mainstream and popular type of pets, such as dogs and cats, rather than the more exotic types.
According to our graph, the median of ratio between time spent and free time of dogs has the highest value, following by cat and guinea. Moreover, we can see that the threshold line is over the rate of 1.0, it means someone define their time for pets as different from free time.
>> cat assignment_data1001.txt | tr ' ' '\n' | sort | uniq

1.0,
10,
11
1-10,
112
12/09/2023,
15
18-19
2
26/08/2023
28.5%
~3.36.
34
5
5,
6
~6.25.
8.34
~8.37
~8.41
a
about
above,
absurd
According
Additionally,
ADHD,
advantages
age
age.
all
allocate
all-round
also
amount
amphibians,
an
and
anim
animal
answer
answers
appearance!
appearance,
appears
appropriate
approximately
arachnids,
are
as
asked
available
average
based
be
because
being
believe
below
benefit
Besides,
between
bond
boxplots,
by
can
care,
care.
cat
category.
cats
cats,
certain
Certain
challenges
cited
citizens.
cleaning
clear
Clearing
code
collected
columns
comes
commitment
companions.
companionship
companionship,
company
compared
concerns.
Conclusion:
correlation
cute
data
data,
data.
Data
define
demonstrated
demonstrating
different
diffferent
disadvantages
disadvantages,
dog
dogs
dogs,
Dogs
due
each
easiest
emphasizing
enjoy
exceptions
excluded
exotic
“factor”
feel
financial
find
first
fish
fishes
fit
following
for
Forms
free
from
From
gathered
gathered,
(Gender,
general
genuinely
giving
Google
graph
graph,
group
guinea
guinea.
Guinea
had
happiness
has
have
have,
having
health.
high
higher
highest
highly
horses
hours
how
However,
(https://forms.gle/LrTkKyHrVUirfZGW9)
idea
if
immense
improving
in
In
indicates
indicating
individual
individuals
investigate
involving
is
issues
issues.
it
it,
its
keep
least
level
Limitations
limited
limits
line
linear
lives.
loneliness.
lower
lowest
main
mainly
mainstream
maintenance
majority
manually.
many
mean
means
median
mental
(money
money
more
Moreover,
most
multiple
next
obscured
of
offer
offering
often
older
on
one
only
opinions
or
organised
other
others
our
over
owner
owner).
owners
participants
people
per
perfect
pet
pet,
pets
pets),
pets,
pets.
pets?
pictures
pig
pigs
plot
popular
positive
public
qualitative
quantitative
Question
questions
questions,
random
rate
rated
rather
rating
ratio
reason
received
relationship
report
reporting
represented
Research
residual
respectively.
respondents
Response
responses
responses,
responses.
rows,
same
satisfaction
satisfaction,
satisfied
scopes
score
scores,
scores.
see
seems
seen
share
showed
showing
shows
signifying
since
solve
someone
source
Source
spend
spent
spent,
still
strong
Structure
students
study
such
suggesting
summary,
support
survey
take
tend
than
that
the
The
their
then
there
these
they
This
threshold
through
time
time,
time.
to
together.
total
type
types
types.
typically
Unfortunately,
university,
upon
using
value
value,
variables
variables.
various
vary,
very
want
was
we
We
week
were
What
when
which
while
While
why
will
willing
with
within
worth
would
year
years
you
>> cat assignment_data1001.txt | tr ' ' '\n' | sort | uniq -c | sort -nr | head -n 10
     41 the
     39 of
     21 to
     19 with
     17 their
     17 and
     15 that
     12 time
     11 a
      9 satisfaction

